src/identities.sql

```diff
@@ -152,22 +152,37 @@ create trigger persons_channel_notify after insert or delete or update on person
 -- cannot drop this since default value of column depends on it
 -- so we always only replace it, unless all tables are dropped
 -- in which case we will recreate the default value anyways
-create or replace function generate_new_posix_id(table_name text, colum_name text)
+create or replace function generate_new_posix_id(
+    table_name text,
+    colum_name text,
+    objects_table text,
+    start_id int default 1000
+)
     returns int as $$
     declare current_max_id int;
     declare new_id int;
+    declare column_exists boolean;
     begin
         execute format('select max(new_data::int) from %I where column_name = %s',
             quote_ident(table_name), quote_literal(colum_name))
             into current_max_id;
         if current_max_id is null then
-            new_id := 1000;
-        elsif current_max_id >= 0 and current_max_id <= 999 then
-            new_id := 1000;
+            new_id := start_id;
+        elsif current_max_id >= 0 and current_max_id <= (start_id - 1) then
+            new_id := start_id;
         else
             new_id := current_max_id + 1;
         end if;
-        return new_id;
+        loop
```

Comment:
I don't think a loop is necessary here, since there is no looping through anything. It should be sufficient with `execute format() into ...`